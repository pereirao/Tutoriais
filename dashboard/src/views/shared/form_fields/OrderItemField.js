// @flow

import React, {PureComponent} from 'react';
import {Dropdown} from 'semantic-ui-react';
import {order_helpers} from '../../../business/order';
import style_sheet from './form_fields.module.css';

const {productName} = order_helpers;

type OrderItemFieldProps = { onChangeText: (string, Object) => void, options: Array<Object> };

class OrderItemField extends PureComponent {
  props: OrderItemFieldProps

  static defaultProps = {name: 'order_item'}
  input: ReactClass<*>

  onChange = (_e, {value}) => {
    this.props.onChangeText(value, true);
  }

  render() {
    const {order, ...dropdown_props} = this.props;
    return (
      <Dropdown
        selection
        className={style_sheet.textInput}
        options={order.order_items.map(oi => {
          const name = productName(oi);
          return {text: name, value: name};
        })}
        onChange={this.onChange}
        {...dropdown_props} />
    );
  }
}

export default OrderItemField;
