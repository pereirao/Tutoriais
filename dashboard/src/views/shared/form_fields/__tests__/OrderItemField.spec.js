import React from 'react';

import OrderItemField from '../OrderItemField';
import order_factory from '../../../../business/order/__tests__/order.factory';

it('renders correctly', () => {
  const order = order_factory.build('with_order_items');
  expect(shallow(<OrderItemField order={order} />)).toMatchSnapshot();
});
