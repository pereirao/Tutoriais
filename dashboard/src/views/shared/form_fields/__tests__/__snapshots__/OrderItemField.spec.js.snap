// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Dropdown
  additionLabel="Add "
  additionPosition="top"
  className="textInput"
  closeOnBlur={true}
  icon="dropdown"
  minCharacters={1}
  name="order_item"
  noResultsMessage="No results found."
  onChange={[Function]}
  openOnFocus={true}
  options={
    Array [
      Object {
        "text": "Budweiser Light — 6 pack, 12 oz bottles — ASDFGHJKL",
        "value": "Budweiser Light — 6 pack, 12 oz bottles — ASDFGHJKL",
      },
      Object {
        "text": "Cupcake — 750 ml bottle — QWERYUOO",
        "value": "Cupcake — 750 ml bottle — QWERYUOO",
      },
    ]
  }
  renderLabel={[Function]}
  selectOnBlur={true}
  selection={true}
/>
`;
